src/game.zig

```diff
@@ -716,16 +775,31 @@ pub fn update(deltaTime: f64) void { // MARK: update()
 				}
 			}
 			if(KeyBoard.key("backward").value > 0.0) {
-				movementSpeed = @max(movementSpeed, 4)*KeyBoard.key("backward").value;
-				movementDir += forward*@as(Vec3d, @splat(-4*KeyBoard.key("backward").value));
+				if (Player.crouching) {
+					movementSpeed = @max(movementSpeed, 2)*KeyBoard.key("backward").value;
+					movementDir += forward*@as(Vec3d, @splat(-2*KeyBoard.key("backward").value));
+				} else {
+					movementSpeed = @max(movementSpeed, 4)*KeyBoard.key("backward").value;
+					movementDir += forward*@as(Vec3d, @splat(-4*KeyBoard.key("backward").value));
+				}
 			}
 			if(KeyBoard.key("left").value > 0.0) {
-				movementSpeed = @max(movementSpeed, 4*KeyBoard.key("left").value);
-				movementDir += right*@as(Vec3d, @splat(4*KeyBoard.key("left").value));
+				if (Player.crouching) {
+					movementSpeed = @max(movementSpeed, 2)*KeyBoard.key("left").value;
+					movementDir += right*@as(Vec3d, @splat(2*KeyBoard.key("left").value));
+				} else {
+					movementSpeed = @max(movementSpeed, 4)*KeyBoard.key("left").value;
+					movementDir += right*@as(Vec3d, @splat(4*KeyBoard.key("left").value));
+				}
 			}
 			if(KeyBoard.key("right").value > 0.0) {
-				movementSpeed = @max(movementSpeed, 4*KeyBoard.key("right").value);
-				movementDir += right*@as(Vec3d, @splat(-4*KeyBoard.key("right").value));
+				if (Player.crouching) {
+					movementSpeed = @max(movementSpeed, 2)*KeyBoard.key("right").value;
+					movementDir += right*@as(Vec3d, @splat(-2*KeyBoard.key("right").value));
+				} else {
+					movementSpeed = @max(movementSpeed, 4)*KeyBoard.key("right").value;
+					movementDir += right*@as(Vec3d, @splat(-4*KeyBoard.key("right").value));
+				}
```

Comment:
Instead of duplicating all of the movement logic, why not just introduce a parameter for the one single value you are actually changing.